src/modinfo.cpp

```diff
@@ -0,0 +1,414 @@
+/*
+	This file is part of Warzone 2100.
+	Copyright (C) 2024  Warzone 2100 Project
+
+	Warzone 2100 is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	Warzone 2100 is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with Warzone 2100; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include "modinfo.h"
+#include "lib/framework/physfs_ext.h"
+#include "lib/framework/file.h"
+#include "lib/framework/wzpaths.h"
+#include "lib/framework/i18n.h"
+#include "wzjsonhelpers.h"
+#include "campaigninfo.h"
+#include "version.h"
+
+#include <nlohmann/json.hpp>
+
+static WzModCompatibilityResult isWZModVersionCompatible(const WZModInfo& modInfo)
+{
+	const auto tagResult = version_extractVersionNumberFromTag(version_getLatestTag());
+	ASSERT(tagResult.has_value(), "No extractable latest tag?? - Please try re-downloading the latest official source bundle");
+	const TagVer latestTaggedVersion = tagResult.value_or(TagVer());
+
+	// minVersionSupported incompatibility check
+	const auto minVerResult = version_extractVersionNumberFromTag(modInfo.minVersionSupported);
+	if (minVerResult.has_value())
+	{
+		if (latestTaggedVersion < minVerResult.value())
+		{
+			return WzModCompatibilityResult::NOT_COMPATIBLE;
+		}
+		// otherwise, fall-through and do additional checks
+	}
+	else
+	{
+		// input minVersionSupported didn't seem to be a version string...
+		return WzModCompatibilityResult::NOT_COMPATIBLE;
+	}
+
+	// additional checks
+	const auto maxVerTestedResult = version_extractVersionNumberFromTag(modInfo.maxVersionTested);
+	auto buildVersionIdentifier = version_getBuildIdentifierReleaseEnvironment();
+	bool isOnReleaseBuild = (buildVersionIdentifier == "release" || buildVersionIdentifier == "preview");
+	if (isOnReleaseBuild && (latestTaggedVersion == minVerResult.value()))
+	{
+		return WzModCompatibilityResult::OK;
+	}
+	else
+	{
+		// latestTaggedVersion is >= minVersionSupported - check maxVersionTested
+		if (!maxVerTestedResult.has_value())
+		{
+			// no maxVersionTested? - _might_ be compatible
+			return WzModCompatibilityResult::MAYBE_COMPATIBLE;
+		}
+		else
```

Comment:
You can remove this `else` altoghether - this will save one indentation level and enhance readability.